infra/modules/providers/azure/storage-account/README.md

```diff
@@ -0,0 +1,118 @@
+# Azure Storage
+
+This Terraform based `storage-account` module grants templates the ability to configure and deploy cloud `storage containers` along with a `storage account` using Microsoft's _**Azure Storage**_ service.
+
+In addition, this module offers both "authentication" and "authorization" features:
+
+- For "authentication", this module automatically enrolls the deployed `storage account` into Microsoft's _**Managed Identities**_ service.
+
+- For "authorization", this module expects to receive the `object id` of any Azure Resource that will behave as a client of the `storage-account`. Enrollment as an Azure Resource client is achieved through Microsoft's _**Role Based Access Secruity**_ (RBAC) service.
+
+#### _More on Azure Storage_
+
+> "A storage account provides a unique namespace in Azure for your data. Every object that you store in Azure Storage has an address that includes your unique account name. A container organizes a set of blobs, similar to a directory in a file system. A storage account can include an unlimited number of containers, and a container can store an unlimited number of blobs." - Source: Microsoft's [Introduction to Azure Blob Storage](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction)
+
+This module deploys a `storage account` along with a `storage container` in order to satisfy blob storage scenarios which are optimized for storing massive amounts of unstructured data, such as text or binary data.
+
+## Characteristics
+
+An instance of the `storage-account` module deploys the _**Azure Storage**_ service in order to provide templates with the following:
+
+- Ability to deploy a Storage Account automatically enrolled into _**Managed Identities**_ within a single resource group.
+
+- Ability to deploy a Storage Account with a role assigned Azure Resource client.
+
+- Ability to deploy a Storage Container within the Storage Account.
+
+## Definition
+
+Azure Storage definition example:
```

Comment:
Is this the internal implementation of your module? It can be removed from the readme if it is.

Reply:
You're right. There's no value add for having the implementation within the readme. This was removed in the latest commit.